### PR TITLE
Container healthcheck and api stats

### PR DESCRIPTION
**Pull request by pleimer**
Here are the amendments for the 'monitoring container healthchecks and api stats' feature. I included a new module that should go in the 'Advanced Features' section as a markdown file (sorry, no adoc). There are also some adjustments made to the multi-cloud section. All of these features are only available in train/osp16.1.3/STF1.1

### CODE REVIEW

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Monitoring Container Health and API Statuses
+
+Container health assesses the status of each of the OpenStack service containers
+by periodically running a health check script using the docker standard. 
+Most OpenStack services implement a healthcheck that logs issues and returns a
+binary status (healthy or unhealthy). For the OpenStack APIs, the healthchecks 
+query the root endpoint and determine the health based on the response time.
+
+To monitor healthchecks in STF, the collectd-sensubility plugin must be enabled
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:9]:
```suggestion
To monitor healthchecks in {ProjectShort}, the collectd-sensubility plugin must be enabled 
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Monitoring Container Health and API Statuses
+
+Container health assesses the status of each of the OpenStack service containers
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:3]:
```suggestion
Container health assesses the status of each of the {OpenStack} service containers
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+by periodically running a health check script using the docker standard. 
+Most OpenStack services implement a healthcheck that logs issues and returns a
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:5]:
```suggestion
Most {OpenStack} services implement a healthcheck that logs issues and returns a
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+binary status (healthy or unhealthy). For the OpenStack APIs, the healthchecks 
+query the root endpoint and determine the health based on the response time.
+
+To monitor healthchecks in STF, the collectd-sensubility plugin must be enabled 
+and configured to work with the amqp1 protocol. The STF architecture considers
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:10]:
```suggestion
and configured to work with the amqp1 protocol. The {ProjectShort} architecture considers
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+healthcheck results to be events and are thus stored in ElasticSearch.
+
+## Prerequisites
+
+- You have created the stf-connectors.yaml file. For more information, see Configuring OpenStack overcloud for Service Telemetry Framework.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:15]:
```suggestion
- You have created the `stf-connectors.yaml` file. For more information, see xref:configuring-red-hat-openstack-platform-overcloud-for-stf_completing-the-stf-configuration[].
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+- You are using the most current version of OpenStack: Train.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:17]:
```suggestion
- You are using the most current version of {OpenStack} {OpenStackVersion}.
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Configuring collectd-sensubility
+
+Within the stf-connectors.yaml, configure collectd-sensubility to use the amqp1
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:21]:
```suggestion
Within the `stf-connectors.yaml`, configure collectd-sensubility to use the amqp1
```

File: doc-Service-Telemetry-Framework/modules/api-stats-updates.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+transport protocol:
+```yaml
+...
+
+CollectdSensubilityTransport: amqp1
+```
+
+In a multicloud environment, the collectd-sensubility events channel will need 
+to be adjusted in  accordance with the new collectd events address. See the [Multiple cloud configuration](https://infrawatch.github.io/documentation/#configuring-multiple-clouds_advanced-features)
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/api-stats-updates.md:30]:
```suggestion
to be adjusted in accordance with the new collectd events address. For more information, see xref:configuring-multiple-clouds_advanced-features[]
```

**Review by leifmadsen — CHANGES_REQUESTED** [review]
Also need to change doc-Service-Telemetry-Framework/modules/api-stats-updates.md to be an asciidoc file (.adoc). Will need to create appropriate label to be able to xref it in doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc [hunk 9]
```diff
@@ -23,7 +23,7 @@
 [id="creating-openstack-environment-file_{context}"]
 = Creating the OpenStack environment file
 
-To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
+To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. [OSP16.1.3+ ONLY] If you have enabled [Container health and API stats monitoring]( #link to new module included in this PR), the `CollectdSensubilityResultsChannel` parameter will also need to be modified. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:26]: Should probably convert this to a NOTE or IMPORTANT. Also need to make sure we update the `OSP` to `{OpenStack}` and update the missing reference (link) to an `xref` of the appropriate header.

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+* You are using the most current version of {OpenStack} {OpenStackVersion}.
+
+.Procedure
+
+. Open the `stf-connectors.yaml` and edit the `collectd-sensubility` paramater:
++
+----
+CollectdSensubilityTransport: amqp1
+----
+. If you have a multicloud environment, you must configure `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:47]:
```suggestion
. If  your environment has multiple clouds, configure `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+// {context}, which ensures that the module has a unique ID even if it is
+// reused multiple times in a guide.
+//
+// In the title, include nouns that are used in the body text. This helps
+// readers and search engines find information quickly.
+[id="monitoring-container-health-and-api-status_{context}"]
+= Monitoring container health and API status
+
+Container health assesses the status of each of the {OpenStack} service containers
+by periodically running a health check script using the docker standard.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: Leif's comment from previous PR: Should Docker be capitalized? Also we're using podman in 16.1 so I think the more correct term is OCI? (Open Container Initiative)

I'm not actually sure what the correct term is here. @csibbitt do you know?

Comment (JoanneOFlynn2018) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: @mcljot do you know if the correct term is OCI (Open Container Initiative)?

Comment (JoanneOFlynn2018) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: Update, @mcljot and myself both didn't know the answer to this. After some research, I think OCI is the correct term. @csibbitt if you would like to concur, that would be great

And for completeness @leifmadsen, when we refer to Docker and Podman in conceptual text (not commands), yes  the first letters should be capitalizied. Thanks for bringing that up, good to know for future work :)

Comment (JoanneOFlynn2018) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: Update, @mcljot and myself both didn't know the answer to this. After some research, I think OCI is the correct term. @csibbitt if you would like to concur, that would be great

And for completeness @leifmadsen, when we refer to Docker and Podman in conceptual text (not commands), yes  the first letters should be capitalizied. Thanks for bringing that up, good to know for future work :)

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+// it after the module has been published to ensure existing links are not
+// broken.
+//
+// The `context` attribute enables module reuse. Every module's ID includes
+// {context}, which ensures that the module has a unique ID even if it is
+// reused multiple times in a guide.
+//
+// In the title, include nouns that are used in the body text. This helps
+// readers and search engines find information quickly.
+[id="monitoring-container-health-and-api-status_{context}"]
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:23]: @pleimer Hi Paul, 
I converted this file to asciidoc and edited it. Can you perform a review of all of the text in this file? Thanks!

Comment (pleimer) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:23]: I will, thanks for doing the conversion!

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc (hunk 11 above)
Comment (leifmadsen) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: I think we can move forward using OCI. Thanks!

Comment (pleimer) [reply doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:27]: OCI seems like the correct term

**Review by pleimer — COMMENTED** [review]
Everything looks good to me!

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc (hunk 9 above)
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:26]:
```suggestion
To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. 

//[OSP16.1.3+ ONLY] 
[NOTE]
If you have enabled Container health and API stats monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
```

**Review by JoanneOFlynn2018 — CHANGES_REQUESTED** [review]
Have made suggestions but cannot commit them. Submitting review and trying again.

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc [hunk 13]
```diff
@@ -23,7 +23,11 @@
 [id="creating-openstack-environment-file_{context}"]
 = Creating the OpenStack environment file
 
-To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
+To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. 
+
+//[OSP16.1.3+ ONLY]
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:28]: @leifmadsen In the customer portal, we still don't publish z stream releases so the best I can do here is put an include_when_16 on it. In fact, there's no point in that either as master will never be synced with v13 or v16. Thoughts?

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc [hunk 14]
```diff
@@ -23,7 +23,11 @@
 [id="creating-openstack-environment-file_{context}"]
 = Creating the OpenStack environment file
 
-To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
+To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. 
+
+//[OSP16.1.3+ ONLY] 
+[NOTE]
+If you have enabled Container health and API stats monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:30]:
```suggestion
If you have enabled Container health and API stats monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[] and xref:monitoring-container-health-and-api-status_advanced-features[]. 
```

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc (hunk 13 above)
Comment (JoanneOFlynn2018) [reply doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:28]: @leifmadsen 
See step one of the procedure below:
1. Create the `stf-connectors.yaml` file
I think we should tell users where to create the file. Tech support has brought this issue up in other docs.

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc [hunk 15]
```diff
@@ -23,7 +23,11 @@
 [id="creating-openstack-environment-file_{context}"]
 = Creating the OpenStack environment file
 
-To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. For more information, see xref:planning-amqp-address-prefixes_advanced-features[].
+To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme. 
+
+//[OSP16.1.3+ ONLY] 
+[NOTE]
+If you have enabled Container health and API stats monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[] and xref:monitoring-container-health-and-api-status_advanced-features[].
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:30]:
```suggestion
If you have enabled container health and API statistics monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[] and xref:monitoring-container-health-and-api-status_advanced-features[]. 
```

Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:30]:
```suggestion
If you have enabled container health and API status monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:planning-amqp-address-prefixes_advanced-features[] and xref:monitoring-container-health-and-api-status_advanced-features[]. 
```

**Review by JoanneOFlynn2018 — APPROVED** [review]

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+by periodically running a health check script using the OCI (Open Container Initiative) standard.
+Most {OpenStack} services implement a healthcheck that logs issues and returns a
+binary status. For the {OpenStack} APIs, the healthchecks
+query the root endpoint and determine the health based on the response time.
+
+To monitor healthchecks in {ProjectShort}, you must enable and configure the `collectd-sensubility` plugin to work with the `amqp1` protocol. The {ProjectShort} architecture considers healthcheck results to be events and are stored in ElasticSearch.
+
+.Prerequisites
+
+* You have created the `stf-connectors.yaml` file. For more information, see xref:configuring-red-hat-openstack-platform-overcloud-for-stf_completing-the-stf-configuration[].
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:36]:
```suggestion
* You have created the `stf-connectors.yaml` file in your templates directory. For more information, see xref:configuring-red-hat-openstack-platform-overcloud-for-stf_completing-the-stf-configuration[].
```

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+* You are using the most current version of {OpenStack} {OpenStackVersion}.
+
+.Procedure
+
+. Open the `stf-connectors.yaml` and edit the `collectd-sensubility` paramater:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:42]:
```suggestion
. Open the `stf-connectors.yaml` and edit the `collectd-sensubility` parameter:
```

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
++
+----
+CollectdSensubilityTransport: amqp1
+----
+. If  your environment has multiple clouds, configure `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:47]:
```suggestion
. If your environment has multiple clouds, configure `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```

Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:47]:
```suggestion
. If  your environment has multiple clouds, configure the `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+* You are using the most current version of {OpenStack} {OpenStackVersion}.
+
+.Procedure
+
+. Open the `stf-connectors.yaml` and edit the `collectd-sensubility` paramater:
++
+----
+CollectdSensubilityTransport: amqp1
+----
+. If  your environment has multiple clouds, configure the `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:47]:
```suggestion
. If your environment has multiple clouds, configure the `collectd-sensubility` events channel with the new collectd events address. Edit the `stf-connectors.yaml` file:
```

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
++
+----
+CollectdSensubilityResultsChannel: collectd/cloudprefix-notify
+----
+
+.Additional resources
+
+For more information about multi-cloud addresses, see xref:configuring-multiple-clouds_advanced-features[].
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-container-health-and-api-status.adoc:55]:
```suggestion
For more information about multiple cloud addresses, see xref:configuring-multiple-clouds_advanced-features[].
```

File: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc (hunk 13 above)
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file.adoc:28]:
```suggestion
```

**Review by leifmadsen — APPROVED** [review]

**Pull request merged**